File: app/src/main/java/firestore/NoteItemTranslate.java
package firestore;

import java.util.HashMap;
import java.util.Map;

import gb.android.notes2.model.NoteListItem;

public class NoteItemTranslate {

    public static class Fields {
        public final static String TITLE = "title";
        public final static String DATE = "date";
        public final static String TEXT_ID = "textId";
    }

    public static NoteListItem docToNoteListItem(String id, Map<String, Object> doc) {
        return new NoteListItem(id, (String) doc.get(Fields.TITLE), (String) doc.get(Fields.DATE));
    }

    public static Map<String, Object> noteListItemToDoc(NoteListItem noteListItem) {
        Map<String, Object> result = new HashMap<>();

        result.put(Fields.TITLE, noteListItem.getTitle());
        result.put(Fields.DATE, noteListItem.getDate());

        return result;
    }
}

File: app/src/main/java/gb/android/notes2/firestore/NoteItemTranslate.kt
package gb.android.notes2.firestore

import gb.android.notes2.model.NoteListItem
import java.util.*

object NoteItemTranslate {
    @JvmStatic
    fun docToNoteListItem(id: String?, doc: Map<String?, Any?>): NoteListItem {
        return NoteListItem(id!!, doc[Fields.TITLE] as String?, doc[Fields.DATE] as String?)
    }

    @JvmStatic
    fun noteListItemToDoc(noteListItem: NoteListItem): Map<String, Any> {
        val result: MutableMap<String, Any> = HashMap()

        result[Fields.TITLE] = noteListItem.title!!
        result[Fields.DATE] = noteListItem.date!!

        return result
    }

    object Fields {
        const val TITLE = "title"
        const val DATE = "date"
        const val TEXT_ID = "textId"
    }
}